wp-content/themes/goonj-crm/functions.php

```diff
@@ -419,8 +419,67 @@ function goonj_is_volunteer_inducted( $volunteer ) {
 function goonj_custom_message_placeholder() {
 	return '<div id="custom-message" class="ml-24"></div>';
 }
+
 add_shortcode( 'goonj_volunteer_message', 'goonj_custom_message_placeholder' );
 
+function generate_volunteer_button_html($buttonUrl) {
+    return sprintf(
+        '<div class="volunteer-button-container" style="display: flex; justify-content: center; align-items: center; border-style:none; border-width:0px; border-radius:5px;">
+            <a href="%s" class="wp-block-button__link has-white-color has-vivid-red-background-color has-text-color has-background has-link-color wp-element-button" 
+                style="border-style:none; border-width:0px; border-radius:5px">
+                Wish to Volunteer?
+            </a>
+        </div>',
+        esc_url($buttonUrl)
+    );
+}
+
+function goonj_contribution_volunteer_signup_button() {
+
+    $activityId = isset($_GET['activityId']) ? sanitize_text_field($_GET['activityId']) : '';
+
+    $activities = \Civi\Api4\Activity::get(FALSE)
+        ->addSelect('source_contact_id')
+        ->addJoin('ActivityContact AS activity_contact', 'LEFT')
+        ->addWhere('id', '=', $activityId)
+        ->addWhere('activity_type_id:label', '=', 'Material Contribution')
+        ->execute();
+
+    if ($activities->count() === 0) {
+        error_log('No activities found for Activity ID: ' . $activityId); // Log if no activities are found
+        return;
+    }
+
+    // Since we have activities, get the single record
+    $activity = $activities->single();
```

Comment:
If there are more than one activity then `single` will give error
```suggestion
    $activity = $activities->first();
```